server/src/test/java/io/druid/curator/discovery/CuratorDruidLeaderSelectorTest.java

```diff
@@ -0,0 +1,172 @@
+/*
+ * Licensed to Metamarkets Group Inc. (Metamarkets) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. Metamarkets licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.curator.discovery;
+
+import com.metamx.emitter.EmittingLogger;
+import com.metamx.emitter.service.ServiceEmitter;
+import io.druid.curator.CuratorTestBase;
+import io.druid.discovery.DruidLeaderSelector;
+import io.druid.java.util.common.logger.Logger;
+import io.druid.server.DruidNode;
+import io.druid.server.initialization.ServerConfig;
+import org.easymock.EasyMock;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.concurrent.atomic.AtomicReference;
+
+/**
+ */
+public class CuratorDruidLeaderSelectorTest extends CuratorTestBase
+{
+  private static final Logger logger = new Logger(CuratorDruidLeaderSelectorTest.class);
+
+  @Before
+  public void setUp() throws Exception
+  {
+    EmittingLogger.registerEmitter(EasyMock.createNiceMock(ServiceEmitter.class));
+    setupServerAndCurator();
+  }
+
+  @Test(timeout = 5000)
```

Comment:
I think we should up this timeout, on my laptop I saw some runs that took between 5-7s, which led to some test failures

Reply:
hmmm, earlier I did not expect it to take long but i think changing (as per the discussion https://github.com/druid-io/druid/pull/4699#pullrequestreview-58205408 ) the sleep from 1 sec to a random value between 1 sec to 5 sec now can make this test take longer as this test exercises that code path. I will update the timeout to 15 secs which should be good enough.

Reply:
@jon-wei https://github.com/druid-io/druid/pull/4756